src/tools/rbd/action/Bench.cc

```diff
@@ -336,44 +370,40 @@ int do_bench(librbd::Image& image, io_type_t io_type,
       ++cur_ios;
       cur_off += io_size;
 
-      if (read_flag)
+      if (read_flag) {
         read_ops++;
-      else
+      } else {
         write_ops++;
-    }
-
-    if (r < 0) {
-      break;
-    }
+      }
 
-    // Set the thread_offsets of next I/O
-    for (i = 0; i < io_threads; ++i) {
       switch (io_pattern) {
       case IO_PATTERN_RAND:
-        thread_offset[i] = (rand() % (size / io_size)) * io_size;
-        continue;
+        pos = (io_start % io_size) + (rand() % num_chunks) * io_size;
+        break;
       case IO_PATTERN_SEQ:
-        if (off < (seq_chunk_length * io_threads)) {
-          thread_offset[i] += io_size;
-        } else {
-          // thread_offset is adjusted to the chunks unassigned to threads.
-          thread_offset[i] = off + (i * io_size);
```

Comment:
Note strange behavior in old code where remainder (size % seq_chunk_length) is non-zero, this part at the end of image is accessed sequentially after rest of image. New code spaces threads at uneven intervals so whole image is accessed by io_thread sequential streams